resources/design-document.md

```diff
@@ -0,0 +1,206 @@
+- Accepts `GET` requests to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts an email and game ID and returns the corresponding GameLogModel.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.1.1 _Get all GameLogs by player Endpoint_
+- Accepts `GET` requests to `/gamelogs/:`
+- Takes the email from cognito
+- Accepts an email and returns the corresponding list of GameLogModels.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.2 _Delete GameLog Endpoint_
+- Accepts a `DELETE` request to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts a game ID and returns the corresponding Deleted GameLogModel.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.3 _Update GameLog Endpoint_
+- Accepts a `PUT` request to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts data to update a gameLog and returns the updated gameLog.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+    - throws `UnauthorizedOwnerException` if attempted to be updated by an unauthorized user.
+
+### 6.2.4 _Get Stats Endpoint_
+- Accepts `GET` requests to `/gamelogs/:`
+- Takes the email from cognito
+- Accepts an email and returns the overall w/l stats for that person's profile.
+
+### 6.2.5 _Get Hero Stats Endpoint_
+- Accepts `GET` request to `/gamelogs/:hero`
+- Takes the email from cognito
+- Accepts an email and returns the w/l stats for a certain hero for that person's profile.
+
+### 6.2.6 _Get Villain Stats Endpoint_
+- Accepts `GET` request to `/gamelogs/:villain`
+- Takes the email from cognito
+- Accepts an email and returns the w/l stats for a certain villain for that person's profile.
+
+### 6.2.7 _Get Aspect Stats Endpoint_
+- Accepts `GET` request to `/gamelogs/:aspect`
+- Takes the email from cognito
+- Accepts an email and returns the w/l stats for a certain aspect for that person's profile.
+
+
+
+# 7. Tables
+
+_Define the DynamoDB tables you will need for the data your service will use. It may be helpful to first think of what
+objects your service will need, then translate that to a table structure, like with the *`Playlist` POJO* versus the
+`playlists` table in the Unit 3 project._
+
+## 7.1 `gamelogs`
+```
+email // partition key, string
+gameId // sort key, String
+date // string (converted date)
+outcome(w/l) // enum
+aspect // enum
+hero // PlayerCharacter
+villain // PlayerCharacter
+
+```
+
+## 7.2 `playercharacters`
+```
+name // partition key, string
+role // Enum
+```
+
+### 7.2.1 `TotalW/LIndex` GSI table
+```
+email // partition key, String
+outome(w/l) // string
+```
+
+### 7.2.2 `TotalW/LHeroIndex` GSI table
+```
+email // partition key, String
+hero // PlayerCharacter
+outome(w/l) // string
+```
+
+
+# 8. Page storyboard
+
+[Link to mock up board with all slides](https://jamboard.google.com/d/1rF50VtNMG0kGWcXDh3yenRobcF1Klmn7RgN8AbX6IkQ/viewer?f=0)
```

Comment:
Can you either make this public or give read access to charlie.penner@nashvillesoftwareschool.com?